docs/architecture/decisions/0001-organizing-code-under-features.md

```diff
@@ -0,0 +1,154 @@
+  common/
+    application-update
+      update-channels.ts
+      update-is-being-downloaded.injectable.ts
+      ...
+      
+    front-end-routes/
+      preferences/
+        app/
+          app-preferences-route.injectable.ts
+          navigate-to-app-preferences.injectable.ts
+
+  renderer/
+    top-bar/
+      top-bar.tsx (UpdateButton)
+    
+    preferences/
+      application.tsx (UpdateChannel)
+```
+
+## Decision
+
+Start organizing code under features, meaning that all code that feature needs to exist in a single directory. Features might have child features which have dependency to the parent. We want to limit the coupling of features, which means that feature is not allowed to touch any other feature without creating dependency to it. We want to have visibility for dependency structures of features.
+
+1. Code that belongs to the feature lives in the feature directory. If I want to remove the feature, I can just delete the directory.
+2. Code that is shared between features lives in the closest parent feature. E.g. child features are using code from parent features.
+3. Parent features should never depend on child features. If the dependency is required, the child feature is the one that depends on the parent feature.
+
+**Example:**
+
+```
+src/
+  features/
+    application-update/
+      installing-update.test.ts
+      ...
+      
+      main/
+        process-checking-of-updates.injectable.ts
+        ...
+      
+      common/
+        update-channels.ts
+        update-is-being-downloaded.injectable.ts
+        ...
+      
+      child-features
+        tray/
+          installing-update-using-tray.test.ts
+          ...
+          
+          main/
+            check-for-updates-tray-menu-item.injectable.ts
+            ...
+        
+        top-bar/
+          renderer/
+            update-button-top-bar-item.injectable.tsx
+            ...
+          
+        preferences/
+          renderer/
+            update-channels-preference-item.injectable.tsx
+            ...
+          
+        application-menu/
+          main/
+            update-channel-application-menu-item.injectable.tsx
+            ...
+            
+        force-update-modal/
+          renderer/
+            force-update-modal.injectable.tsx
+            ...
+       
+    tray/
+      main/
+        tray-menu-item-injection-token.ts
+        ...
+        
+    top-bar/
+      renderer/
+        top-bar-item-injection-token.ts
+        top-bar.tsx
+        ...
+        
+    preferences/
+      renderer/
+        preference-item-injection-token.ts
+        ...
+        
+      common/
+        app-preferences-route.injectable.ts
+        navigate-to-app-preferences.injectable.ts
+        ...
+        
+      child-features/
+        tray/
+          main/
+            navigate-to-preferences-tray-menu-item.injectable.ts 
+            ...
+```
+
+## Consequences
+
+#### Pros
+1. We will start to focus on the features, not the technical details.
```

Comment:
I guess we should also discuss where technical details should live like "readFileInjectable"?